extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java

```diff
@@ -361,6 +75,35 @@ NativeImageResourceBuildItem resource() {
         return new NativeImageResourceBuildItem("META-INF/services/atlas/module/atlas.module");
     }
 
+    @BuildStep
+    void addDependencies(BuildProducer<IndexDependencyBuildItem> indexDependency) {
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-core"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-xml-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-java-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-json-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-csv-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-dfdl-model"));
+    }
+
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection(CombinedIndexBuildItem combinedIndex) {
+        IndexView index = combinedIndex.getIndex();
+        // register Atlasmap model classes for serialize/deserialize
+        String[] dtos = index.getKnownClasses().stream().map(ci -> ci.name().toString())
+                .filter(n -> n.startsWith("io.atlasmap.v2")
+                        || n.startsWith("io.atlasmap.actions")
+                        || n.startsWith("io.atlasmap.converters")
+                        || n.startsWith("io.atlasmap.json.v2")
+                        || n.startsWith("io.atlasmap.java.v2")
+                        || n.startsWith("io.atlasmap.xml.v2")
+                        || n.startsWith("io.atlasmap.csv.v2")
+                        || n.startsWith("io.atlasmap.dfdl.v2"))
```

Comment:
I think it is fine for the moment to filter base on the package but wonder if we can switch to a more efficient way, like registering for reflection based on interfaces and SPIs.

Reply:
How can I do that? I don't know how to look for it. Some of the classes extends an interface, and some just Serializable. And some needed Enums do not extend anything.
Also I was wondering if it's doable to split the Atlasmap into different packages, so that end user chooses if he/she needs java, XML, CSV ... Because we do register lot of classes for maybe just one type of conversion.

Reply:
I think we need to explore using a mix of things, like to use `IndexView.getAllKnownImplementors` and/or `IndexView.getAllKnownSubclasses` for most of the bits and everything else as a special case. 

We could also use a mix of strategies like searching also from all the Serializable and enums withting the `io.atlasmap` package. I'd also explore the option to contribute to Atlasmap to make discovery easier.

I'm +1 to explore if we can split atlasmap extension so it's up to the user to decide what to include and we could also think about having an uber extension that includes everything as today (as transitive dep).